### PR TITLE
Update the consensus bundles

### PR DESCRIPTION
**Pull request by dnadales**
This patch divides the `ouroboros-consensus` and
`ouroboros-consensus-packages` into library and test bundles:

- Creates the `ouroboros-consensus-test` bundle, containing:
  - `ouroboros-consensus-test`
  - `ouroboros-consensus-mock-test`
  - `ouroboros-consensus-tutorials`
- Creates the `ouroboros-consensus-cardano-test` bundle, containing:
  - `ouroboros-consensus-byronspec`
  - `ouroboros-consensus-cardano-test`
  - `ouroboros-consensus-byron-test`
  - `ouroboros-consensus-shelley-test`
- Updates the `ouroboros-consensus` and `ouroboros-consensus-packages` bundles by removing the test packages.

Bundle update and creation is done by adding `changelog.d` symlinks and creating or updating the `CHANGELOG.md` files inside each package.

# Checklist

- Branch
    - [ ] Commit sequence broadly makes sense
    - [ ] Commits have useful messages
    - [ ] The documentation has been properly updated
    - [ ] New tests are added if needed and existing tests are updated
    - [ ] Any changes affecting Consensus packages must have an entry in the appropriate `changelog.d` directory created using [`scriv`](https://github.com/input-output-hk/scriv). If in doubt, see the [Consensus release process](../ouroboros-consensus/docs/ReleaseProcess.md).
    - [ ] Any changes in the Consensus API (every exposed function, type or module) that has changed its name, has been deleted, has been moved, or altered in some other significant way must leave behind a `DEPRECATED` warning that notifies downstream consumers. If deprecating a whole module, remember to add it to `./scripts/ci/check-stylish.sh` as otherwise `stylish-haskell` would un-deprecate it.
    - [ ] If this branch changes Network and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../docs/interface-CHANGELOG.md)
    - [ ] If serialization changes, user-facing consequences (e.g. replay from genesis) are confirmed to be intentional.
- Pull Request
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description at least containing the following information:
      - What does this PR change?
      - Why these changes were needed?
      - How does this affect downstream repositories and/or end-users?
      - Which ticket does this PR close (if any)? If it does, is it [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)?
    - [ ] Reviewer requested

### CODE REVIEW

File: ouroboros-consensus/CHANGELOG.md [hunk 1]
```diff
@@ -2,12 +2,9 @@
 
 This Changelog describes changes on the ledger agnostic Consensus packages:
 - `ouroboros-consensus`
-- `ouroboros-consensus-test`
 - `ouroboros-consensus-diffusion`
 - `ouroboros-consensus-mock`
```
[the remark this answers is not in the document]
Comment (dnadales) [reply ouroboros-consensus/CHANGELOG.md:6]: Yup. @Jasagredo addressed this [here](https://github.com/input-output-hk/ouroboros-network/pull/4450#pullrequestreview-1336916678)

File: ouroboros-consensus-cardano/CHANGELOG.md [hunk 2]
```diff
@@ -2,12 +2,8 @@
 
 This Changelog describes changes on the Cardano specific Consensus packages:
 - `ouroboros-consensus-byron`
-- `ouroboros-consensus-byron-test`
-- `ouroboros-consensus-byronspec`
 - `ouroboros-consensus-shelley`
-- `ouroboros-consensus-shelley-test`
 - `ouroboros-consensus-cardano`
-- `ouroboros-consensus-cardano-test`
 - `ouroboros-consensus-cardano-tools`
```
[the remark this answers is not in the document]
Comment (dnadales) [reply ouroboros-consensus-cardano/CHANGELOG.md:7]: See https://github.com/input-output-hk/ouroboros-network/pull/4450#pullrequestreview-1336916678

**Review by Jasagredo — APPROVED** [review]

**Pull request merged**